storage/client/queue/src/main/java/com/azure/storage/queue/QueueServiceClientBuilder.java

```diff
@@ -230,11 +233,28 @@ public QueueServiceClientBuilder credential(SharedKeyCredential credential) {
      * @throws NullPointerException If {@code connectionString} is {@code null}.
      */
     public QueueServiceClientBuilder connectionString(String connectionString) {
-        Objects.requireNonNull(connectionString);
         this.sharedKeyCredential = SharedKeyCredential.fromConnectionString(connectionString);
+        getEndPointFromConnectionString(connectionString);
         return this;
     }
 
+    private void getEndPointFromConnectionString(String connectionString) {
```

Comment:
Also, consider turning this into a utility method as parsing the connection string to get key-value pairs may be used in multiple places.

Reply:
This will move to common utility after we extract common module. It will address all these pain.
I am doing this just to minimize the the PR's scope.

Reply:
Please refer this one: https://github.com/Azure/azure-sdk-for-java/pull/4658